File: src/com/filewatcherapp/models/FileEvent.java
package com.filewatcherapp.models;

import java.time.LocalDateTime;

/**
 * Represents a file system event, including file name, extension, full path, event type, and timestamp.
 */
public class FileEvent {
    /** File name without directory path. */
    private String fileName;

    /** File extension, e.g. ".txt". */
    private String extension;

    /** Full path to the file where the event occurred. */
    private String filePath;

    /** Type of file system event: CREATED, DELETED, MODIFIED, RENAMED. */
    private String eventType;

    /** Date and time when the event occurred. */
    private LocalDateTime timestamp;

    /**
     * Constructs a new FileEvent instance.
     *
     * @param fileName  Name of the file.
     * @param extension File extension.
     * @param filePath  Full file path.
     * @param eventType Type of event.
     * @param timestamp Timestamp of the event.
     */
    public FileEvent(String fileName, String extension, String filePath, String eventType, LocalDateTime timestamp) {
        this.fileName = fileName;
        this.extension = extension;
        this.filePath = filePath;
        this.eventType = eventType;
        this.timestamp = timestamp;
    }

    // Getters and Setters

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getExtension() {
        return extension;
    }

    public void setExtension(String extension) {
        this.extension = extension;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public String getEventType() {
        return eventType;
    }

    public void setEventType(String eventType) {
        this.eventType = eventType;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "FileEvent{" +
                "fileName='" + fileName + '\'' +
                ", extension='" + extension + '\'' +
                ", filePath='" + filePath + '\'' +
                ", eventType='" + eventType + '\'' +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: FilesystemWatcher/Model/FileEvent.cs
namespace FilesystemWatcher.Model
{
    /// <summary>
    /// Represents a single file system event, including its name, path, type, and timestamp.
    /// </summary>
    /// <author>Mansur Yassin</author>
    /// <author>Tairan Zhang</author>
    public class FileEvent
    {
        /// <summary>
        /// Gets or sets the name of the file.
        /// </summary>
        public string FileName { get; set; } = "";

        /// <summary>
        /// Gets or sets the file extension (including the leading dot).
        /// </summary>
        public string Extension { get; set; } = "";

        /// <summary>
        /// Gets or sets the full path to the file.
        /// </summary>
        public string FilePath { get; set; } = "";

        /// <summary>
        /// Gets or sets the type of event (e.g., "Created", "Changed", "Deleted", "Renamed", "Manual").
        /// </summary>
        public string EventType { get; set; } = "";

        /// <summary>
        /// Gets or sets the timestamp when the event occurred.
        /// </summary>
        public DateTime Timestamp { get; set; }
    }
}